test/expr/capture/local_lazy.swift

```diff
@@ -0,0 +1,16 @@
+// RUN: %target-swift-frontend -emit-silgen -verify %s
+struct S {
+  func foo() -> Int {
+    // Make sure the decl context for the autoclosure passed to ?? is deep
+    // enough that it can 'see' the capture of $0 from the outer closure.
+    lazy var nest: (Int) -> Int = { Optional<Int>.none ?? $0 }
```

Comment:
Can you FileCheck the SIL function definitions to ensure they have capture arguments too, just in case?

Reply:
Followed the other tests and checked the AST dump. The SILGen test is now being used to make sure we don't crash.